src/hotspot/share/gc/g1/g1ParScanThreadState.cpp

```diff
@@ -267,7 +277,10 @@ void G1ParScanThreadState::start_partial_objarray(G1HeapRegionAttr dest_attr,
     push_on_queue(ScannerTask(PartialArrayScanTask(from_obj)));
   }
 
-  G1ScanInYoungSetter x(&_scanner, dest_attr.is_young());
+  // FIXME: check if we could just use dest_attr
+  dest_attr = _g1h->region_attr(to_array);
+  assert(!_g1h->region_attr(to_array).is_in_cset(), "must not scan object from cset here");
+  G1SkipCardEnqueueSetter x(&_scanner, dest_attr.is_new_survivor() || dest_attr.is_in_cset());
```

Comment:
Is this still WIP?

Reply:
No, but has the same bug as the other similar place.